File: Cpp/Arrays & Hashing/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int longest = 0;
        unordered_set<int> numsSet(nums.begin(), nums.end());

        for (int num : numsSet) {
            if (!numsSet.count(num - 1)) {
                int length = 1;
                while (numsSet.count(num + length)) {
                    length++;
                }
                longest = max(length, longest);
            }
        }

        return longest;
    }
};

File: Python/Arrays & Hashing/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        longest = 0
        nums_set = set(nums)

        for num in nums_set:
            if (num - 1) not in nums_set:
                length = 1
                while(num + length) in nums_set:
                    length += 1
                longest = max(length, longest)
        
        return longest
